### PR TITLE
chore: integration events store

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-2437/create-new-integration-event-store

Adds a new `IntegrationEventsStore`.

### CODE REVIEW

**Review by sighphyre — APPROVED** [review]
Ye, think it's good

**Review by chriswk — APPROVED** [review]

File: src/lib/openapi/spec/integration-event-schema.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            example: 42,
+        },
+        createdAt: {
+            type: 'string',
+            format: 'date-time',
+            description:
+                'The date and time of when the integration event was created. In other words, the date and time of when the integration handled the event.',
+            example: '2023-12-27T13:37:00+01:00',
+        },
+        state: {
```
[the remark this answers is not in the document]
Comment (nunogois) [reply src/lib/openapi/spec/integration-event-schema.ts:41]: Is this really worth creating a new migration to rename this column in the database table?

File: src/lib/openapi/spec/integration-event-schema.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            description:
+                'The state of the integration event. Can be one of `success`, `failed` or `successWithErrors`.',
+            example: 'failed',
+        },
+        stateDetails: {
+            type: 'string',
+            description: 'Details about the state of the integration event.',
+            example: 'Status code: 429 - Rate limit reached.',
+        },
+        event: {
```
[the remark this answers is not in the document]
Comment (nunogois) [reply src/lib/openapi/spec/integration-event-schema.ts:53]: >If the long term goal is to have things like terraform and Jira also work with this feature

I don't think it is. I believe this project should only handle integrations that are outgoing in their communication. Can you help clarify this @chriswk?

File: src/lib/features/integration-events/integration-events.e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        10,
+        0,
+    );
+
+    expect(events).toHaveLength(2);
+    expect(events[0].state).toBe('failed');
+    expect(events[1].state).toBe('success');
+});
+
+test('paginate to latest event', async () => {
```
[the remark this answers is not in the document]
Comment (nunogois) [reply src/lib/features/integration-events/integration-events.e2e.test.ts:116]: Do you mean something like this? https://github.com/Unleash/unleash/pull/7613/commits/82bb3a099c9a4d2c824764e18908ec17b3b0206f

File: src/lib/openapi/spec/integration-event-schema.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sighphyre) [reply src/lib/openapi/spec/integration-event-schema.ts:41]: > Is this really worth creating a new migration to rename this column in the database table?

😭 no I don't it's worth it

File: src/lib/features/integration-events/integration-events.e2e.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sighphyre) [reply src/lib/features/integration-events/integration-events.e2e.test.ts:116]: Ye, love it

**Pull request merged**